src/middleware/api.js

```diff
@@ -31,24 +38,38 @@ export function addExpandersToPath(path, type) {
   const { settings } = config;
   const { apiExpanders = [] } = settings;
 
-  const pathPart = path.split('?')[0] || '';
-  const expanders = apiExpanders
-    .filter((expand) => matchPath(pathPart, expand.match) && expand[type])
+  const {
+    url,
+    query: { expand, ...query },
+  } = qs.parseUrl(path);
+
+  const expandersFromConfig = apiExpanders
+    .filter((expand) => matchPath(url, expand.match) && expand[type])
     .map((expand) => expand[type]);
-  const query = qs.parse(qs.extract(path));
-  const expand = compact(union([query.expand, ...flatten(expanders)]));
-  if (expand) {
-    query.expand = expand;
-  }
+
+  const expandMerge = compact(union([expand, ...flatten(expandersFromConfig)]));
+
+  const stringifiedExpand = qs.stringify(
+    { expand: expandMerge },
+    {
+      arrayFormat: 'comma',
+      encode: false,
+    },
+  );
+
   const stringifiedQuery = qs.stringify(query, {
-    arrayFormat: 'comma',
     encode: false,
   });
-  if (!stringifiedQuery) {
-    return pathPart;
-  }
 
-  return `${pathPart}?${stringifiedQuery}`;
+  if (stringifiedQuery && stringifiedExpand) {
+    return `${url}?${stringifiedExpand}&${stringifiedQuery}`;
+  } else if (!stringifiedQuery && stringifiedExpand) {
```

Comment:
Nice. We were indeed missing these conditions which defer queries and expanders working together.